packages/flutter/lib/src/rendering/copy_interceptor.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'selection.dart';
+
+/// Typedef for the inline intercept callback.
+typedef CopyInterceptorCallback = String Function(List<SelectedContent> selections);
+
+/// A handler for intercepting copy operations.
+abstract class CopyInterceptor {
+  /// Creates a [CopyInterceptor].
+  const CopyInterceptor();
+
+  /// Creates a [CopyInterceptor] that joins selections with the given [separator].
+  const factory CopyInterceptor.separator(String separator) = _SeparatorCopyInterceptor;
+
+  /// Creates a [CopyInterceptor] that uses the given [intercept] callback.
+  const factory CopyInterceptor.inline(CopyInterceptorCallback intercept) = _InlineCopyInterceptor;
+
+  /// A [CopyInterceptor] that joins selections with an empty string.
+  static const CopyInterceptor none = CopyInterceptor.separator('');
+
+  /// A [CopyInterceptor] that joins selections with a newline character.
+  static const CopyInterceptor newline = CopyInterceptor.separator('\n');
+
+  /// A [CopyInterceptor] that joins selections with a space character.
+  static const CopyInterceptor space = CopyInterceptor.separator(' ');
+
+  /// Convert the given [selections] to a single string.
+  String intercept(List<SelectedContent> selections);
+}
+
+class _SeparatorCopyInterceptor extends CopyInterceptor {
+  const _SeparatorCopyInterceptor(this.separator);
+
+  final String separator;
+
+  @override
+  String intercept(List<SelectedContent> selections) {
+    return selections.map((SelectedContent selection) => selection.plainText).join(separator);
+  }
+}
+
+class _InlineCopyInterceptor extends CopyInterceptor {
+  const _InlineCopyInterceptor(this._interceptInline);
+
+  final CopyInterceptorCallback _interceptInline;
+
+  @override
+  String intercept(List<SelectedContent> selections) => _interceptInline(selections);
+}
```

Comment:
What if a user wants to do something like this: every Row widget copies in a special way, and every other widget copies in a different way? Row is just an example here, could be any specific widget. I'm wondering whether this approach will give users this kind of flexibility or not.

Reply:
This PR adds the ability to nest SelectionAreas so there is no reason you couldn't have whatever behavior you want